build.gradle

```diff
@@ -47,6 +47,9 @@ repositories {
 }
 
 dependencies {
+    // We should not include javax.mail in the dependencies as it is included with App Engine SDK and adding it will cause
+    // conflicts. However, in a very unlikely scenario that it is required, we should repackage it or use other workarounds.
+    // See https://github.com/TEAMMATES/teammates/pull/7525 for more info.
```

Comment:
Agree that more documentation is better, and adding a doc about dependencies sounds good.

For this particular instance, however, I feel the possibility of someone adding back `javax.mail` is very unlikely (our dependencies don't often change; also the mail library hasn't been touched since the beginning of CI in TEAMMATES). Furthermore, the comment is about something that has been removed from, not something that exists in, the codebase. It's like dead code (but it's a dead comment); hence, I'd still prefer it to be removed.